### DISCUSSION TITLE
HomeBridge Integration Mini Guide

### DISCUSSION THREAD

**Discussion by milmber** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#710] [Show and tell]
HomeBridge Integration Mini Guide based on this post https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/issues/109#issuecomment-900157680  and example from @PierreBier

1. Install the [Aedes embedded MQTT Broker](https://github.com/kevinkub/homebridge-aedes) plugin in HomeBridge
2. Configure the Aedes plugin  as follows:
```json
{
    "name": "Aedes MQTT",
    "host": "YOUR_HOMEBRIDGE_SERVER_IP_ADDRESS",
    "auth": {
        "isEnabled": true,
        "username": "YOUR_SECRET_USERNAME",
        "password": "YOUR_SECRET_PASSWORD"
    },
    "platform": "HomebridgeAedes"
}
```
3. Install the [homebridge-mqttthing](https://github.com/arachnetech/homebridge-mqttthing) plugin
4. Configure the homebridge-mqttthing as follows under the "accessories" section of your configuration file 
```json   
"accessories": [
        {
            "type": "temperatureSensor",
            "name": "LazySPA Temperature",
            "url": "mqtt://YOUR_HOMEBRIDGE_SERVER_IP_ADDRESS:1883",
            "username": "YOUR_SECRET_USERNAME",
            "password": "YOUR_SECRET_PASSWORD",
            "logMqtt": true,
            "topics": {
                "getCurrentTemperature": {
                    "topic": "layzspa/message",
                    "apply": "return JSON.parse(message).TMP;"
                }
            },
            "accessory": "mqttthing"
        },
       {
            "type": "thermostat",
            "name": "LazySPA Heating",
            "url": "mqtt://YOUR_HOMEBRIDGE_SERVER_IP_ADDRESS:1883",
            "username": "YOUR_SECRET_USERNAME",
            "password": "YOUR_SECRET_PASSWORD",
            "logMqtt": true,
            "integerValue": true,
            "topics": {
                "getCurrentHeatingCoolingState": {
                    "topic": "layzspa/message",
                    "apply": "return JSON.parse(message).RED == 0 ? '0' : '1';"
                },
                "getCurrentTemperature": {
                    "topic": "layzspa/message",
                    "apply": "return JSON.parse(message).TMP;"
                },
                "getTargetTemperature": {
                    "topic": "layzspa/message",
                    "apply": "return JSON.parse(message).TGT;"
                },
                "setTargetTemperature": {
                    "topic": "layzspa/command",
                    "apply": "return JSON.stringify({CMD:0,VALUE:message,XTIME:0,INTERVAL:0})"
                },
                "setTargetHeatingCoolingState": {
                    "topic": "layzspa/command",
                    "apply": "return JSON.stringify({CMD:3,VALUE:message,XTIME:0,INTERVAL:0})"
                }
            },
            "minTemperature": 20,
            "maxTemperature": 40,
            "heatingCoolingStateValues": [
                "0",
                "1"
            ],
            "restrictHeatingCoolingState": [
                0,
                1
            ],
            "accessory": "mqttthing"
        },
        {
            "type": "switch",
            "name": "LazySPA Pump",
            "url": "mqtt://YOUR_HOMEBRIDGE_SERVER_IP_ADDRESS:1883",
            "username": "YOUR_SECRET_USERNAME",
            "password": "YOUR_SECRET_PASSWORD",
            "logMqtt": true,
            "integerValue": true,
            "offValue": "return JSON.stringify({CMD:4,VALUE:0,XTIME:0,INTERVAL:0})",
            "topics": {
                "getOnline": {
                    "topic": "layzspa/message",
                    "apply": "return JSON.parse(message).PWR;"
                },
                "getOn": {
                    "topic": "layzspa/message",
                    "apply": "return JSON.parse(message).FLT;"
                },
                "setOn": {
                    "topic": "layzspa/command",
                    "apply": "return JSON.stringify({CMD:4,VALUE:message,XTIME:0,INTERVAL:0})"
                }
            },
            "accessory": "mqttthing"
        },
        {
            "type": "switch",
            "name": "LazySPA Bubbles",
            "url": "mqtt://YOUR_HOMEBRIDGE_SERVER_IP_ADDRESS:1883",
            "username": "YOUR_SECRET_USERNAME",
            "password": "YOUR_SECRET_PASSWORD",
            "logMqtt": true,
            "integerValue": true,
            "offValue": "return JSON.stringify({CMD:2,VALUE:0,XTIME:0,INTERVAL:0})",
            "topics": {
                "getOnline": {
                    "topic": "layzspa/message",
                    "apply": "return JSON.parse(message).PWR;"
                },
                "getOn": {
                    "topic": "layzspa/message",
                    "apply": "return JSON.parse(message).AIR;"
                },
                "setOn": {
                    "topic": "layzspa/command",
                    "apply": "return JSON.stringify({CMD:2,VALUE:message,XTIME:0,INTERVAL:0})"
                }
            },
            "accessory": "mqttthing"
        },
        {
            "type": "switch",
            "name": "LazySPA Jet",
            "url": "mqtt://YOUR_HOMEBRIDGE_SERVER_IP_ADDRESS:1883",
            "username": "YOUR_SECRET_USERNAME",
            "password": "YOUR_SECRET_PASSWORD",
            "logMqtt": true,
            "integerValue": true,
            "offValue": "return JSON.stringify({CMD:11,VALUE:0,XTIME:0,INTERVAL:0})",
            "topics": {
                "getOnline": {
                    "topic": "layzspa/message",
                    "apply": "return JSON.parse(message).PWR;"
                },
                "getOn": {
                    "topic": "layzspa/message",
                    "apply": "return JSON.parse(message).HJT;"
                },
                "setOn": {
                    "topic": "layzspa/command",
                    "apply": "return JSON.stringify({CMD:11,VALUE:message,XTIME:0,INTERVAL:0})"
                }
            },
            "accessory": "mqttthing"
        }
```
5. If your Lazy SPA model doesn't have a HydroJet function, remove the "LazySPA Jet" section of the configuration file.
6. Log into the ESP8266 Wifi Remote and Configure the MQTT Settings as follows
![Screenshot 2024-05-19 at 12 09 13](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/7280701/885f1a97-e7a7-49b4-b379-de15b172674b)

**Comment by frizby66** [comment]
Hi, currently the Homebridge Accessories doesn't appear to read back the state of the current set temperature. I can see it coming through in the matt string, but its not be reflected in the accessory button in home bridge.

`[08/07/2025, 14:11:50] [LazySPA Heating] Received MQTT: layzspa/message = {"CONTENT":"STATES","TIME":1751980310,"LCK":1,"PWR":1,"UNT":1,"AIR":0,"GRN":1,"RED":0,"FLT":1,"CH1":32,"CH2":51,"CH3":57,"HJT":0,"BRT":7,"ERR":0,"GOD":0,"TGT":35,"TMP":39,"VTMC":38.00999832,"VTMF":100.4179993,"AMBC":10,"AMBF":50,"AMB":10,"VTM":38.00999832,"TGTC":35,"TMPC":39,"TGTF":95,"TMPF":102}
[08/07/2025, 14:11:50] [LazySPA Heating] apply() function decoded message to [0]
[08/07/2025, 14:11:50] [LazySPA Heating] apply() function decoded message to [39]
[08/07/2025, 14:11:50] [LazySPA Heating] apply() function decoded message to [35]
[08/07/2025, 14:11:50] [LazySPA Pump] Received MQTT: layzspa/message = {"CONTENT":"STATES","TIME":1751980310,"LCK":1,"PWR":1,"UNT":1,"AIR":0,"GRN":1,"RED":0,"FLT":1,"CH1":32,"CH2":51,"CH3":57,"HJT":0,"BRT":7,"ERR":0,"GOD":0,"TGT":35,"TMP":39,"VTMC":38.00999832,"VTMF":100.4179993,"AMBC":10,"AMBF":50,"AMB":10,"VTM":38.00999832,"TGTC":35,"TMPC":39,"TGTF":95,"TMPF":102}
[08/07/2025, 14:11:50] [LazySPA Pump] apply() function decoded message to [1]
[08/07/2025, 14:11:50] [LazySPA Pump] apply() function decoded message to [1]
[08/07/2025, 14:12:00] [LazySPA Bubbles] Received MQTT: layzspa/message = {"CONTENT":"STATES","TIME":1751980320,"LCK":1,"PWR":1,"UNT":1,"AIR":0,"GRN":1,"RED":0,"FLT":1,"CH1":32,"CH2":51,"CH3":57,"HJT":0,"BRT":7,"ERR":0,"GOD":0,"TGT":35,"TMP":39,"VTMC":38.00999832,"VTMF":100.4179993,"AMBC":10,"AMBF":50,"AMB":10,"VTM":38.00999832,"TGTC":35,"TMPC":39,"TGTF":95,"TMPF":102}
[08/07/2025, 14:12:00] [LazySPA Bubbles] apply() function decoded message to [0]
[08/07/2025, 14:12:00] [LazySPA Bubbles] apply() function decoded message to [1]
[08/07/2025, 14:12:00] [LazySPA Temperature] Received MQTT: layzspa/message = {"CONTENT":"STATES","TIME":1751980320,"LCK":1,"PWR":1,"UNT":1,"AIR":0,"GRN":1,"RED":0,"FLT":1,"CH1":32,"CH2":51,"CH3":57,"HJT":0,"BRT":7,"ERR":0,"GOD":0,"TGT":35,"TMP":39,"VTMC":38.00999832,"VTMF":100.4179993,"AMBC":10,"AMBF":50,"AMB":10,"VTM":38.00999832,"TGTC":35,"TMPC":39,"TGTF":95,"TMPF":102}
[08/07/2025, 14:12:00] [LazySPA Temperature] apply() function decoded message to [39]
[08/07/2025, 14:12:00] [LazySPA Heating] Received MQTT: layzspa/message = {"CONTENT":"STATES","TIME":1751980320,"LCK":1,"PWR":1,"UNT":1,"AIR":0,"GRN":1,"RED":0,"FLT":1,"CH1":32,"CH2":51,"CH3":57,"HJT":0,"BRT":7,"ERR":0,"GOD":0,"TGT":35,"TMP":39,"VTMC":38.00999832,"VTMF":100.4179993,"AMBC":10,"AMBF":50,"AMB":10,"VTM":38.00999832,"TGTC":35,"TMPC":39,"TGTF":95,"TMPF":102}
[08/07/2025, 14:12:00] [LazySPA Heating] apply() function decoded message to [0]
[08/07/2025, 14:12:00] [LazySPA Heating] apply() function decoded message to [39]
[08/07/2025, 14:12:00] [LazySPA Heating] apply() function decoded message to [35]`
